### PR TITLE
Add Python 3.12 to supported versions

### PR DESCRIPTION
**Pull request by JacobCallahan**
Python 3.12 is out and should be supported by Robottelo. Also switched the weekly workflow to use Python 3.12 instead of 3.11.

### CODE REVIEW

**Comment by jyejare** [comment]
@JacobCallahan Dependencies installation is failing for 6.12 in configparser.

**Comment by JacobCallahan** [comment]
Blocked on https://github.com/ParallelSSH/ssh2-python/issues/193

**Comment by JacobCallahan** [comment]
Removing stale since there are now updates on https://github.com/SatelliteQE/broker/pull/251

**Comment by JacobCallahan** [comment]
trigger: test-robottelo
broker: 251
pytest: tests/foreman/ -k 'test_host_registration_end_to_end or test_positive_erratum_applicability or test_positive_upload_content'

**Comment by JacobCallahan** [comment]
trigger: test-robottelo
broker: 251
pytest: tests/foreman/ -k 'test_host_registration_end_to_end or test_positive_erratum_applicability or test_positive_upload_content'

**Comment by JacobCallahan** [comment]
Out of the 26 tests ran by PRT, 23 passed. The 3 that failed did so due to errors unrelated to Broker itself. Will proceed with a new Broker release and update this PR after.

**Comment by JacobCallahan** [comment]
It looks like there were a number of long-deprecated changes in unittest that have been removed in python 3.12

**Comment by JacobCallahan** [comment]
trigger: test-robottelo
broker: 251
pytest: tests/foreman/ -k 'test_host_registration_end_to_end or test_positive_erratum_applicability or test_positive_upload_content'

**Comment by JacobCallahan** [comment]
trigger: test-robottelo
pytest: tests/foreman/ -k end_to_end

**Comment by JacobCallahan** [comment]
Looking through the results, I don't see any failures attributable to this PR.

**Review by jyejare — APPROVED** [review]
Looks Good!

File: requirements.txt [hunk 1]
```diff
@@ -30,4 +30,9 @@ wrapanapi==3.6.0
 # Get airgun, nailgun and upgrade from master
 git+https://github.com/SatelliteQE/airgun.git@master#egg=airgun
 git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun
+# Broker currently is unable to push to PyPi due to [1] and [2]
+# In the meantime, we install directly from the repo
+# [1] - https://github.com/ParallelSSH/ssh2-python/issues/193
+# [2] - https://github.com/pypi/warehouse/issues/7136
```
Comment (Gauravtalreja1) [inline requirements.txt:36]: I think both of these issues have been resolved now, so can we keep this the way it is?

File: robottelo/cli/base.py [hunk 2]
```diff
@@ -132,26 +132,26 @@ def delete(cls, options=None, timeout=None):
...
-    def delete_parameter(cls, options=None):
+    def delete_parameter(cls, options=None, timeout=None):
         """
         Deletes parameter from record.
         """
 
         cls.command_sub = 'delete-parameter'
 
-        result = cls.execute(cls._construct_command(options))
+        result = cls.execute(cls._construct_command(options), ignore_stderr=False, timeout=timeout)
```
Comment (Gauravtalreja1) [inline robottelo/cli/base.py:142]: is this change intended here?

File: requirements.txt (hunk 1 above)
Comment (ogajduse) [reply requirements.txt:36]: @Gauravtalreja1 https://github.com/ParallelSSH/ssh2-python/issues/193 is still open.

**Review by ogajduse — APPROVED** [review]
All changes LGTM. :rocket:

File: robottelo/cli/base.py (hunk 2 above)
Comment (JacobCallahan) [reply robottelo/cli/base.py:142]: it is, yes. similar commands had been updated with these arguments, and the lack of them ended up with failing mock tests.

**Review by Gauravtalreja1 — APPROVED** [review]
@JacobCallahan LGTM, Thank you for adding this! :heart:
